dashboard/forms.py

```diff
@@ -200,6 +214,16 @@ def include_extract_form(dg):
     else:
         return False
 
+def include_clean_comp_data_form(dg):
+    '''Returns the CleanCompDataForm based on conditions of DataGroup
+    type = Composition and at least 1 document extracted
+    '''
+    if not dg.type in ['CO']:
+        return False
+    if dg.extracted_docs() > 0:
```

Comment:
```suggestion
    if dg.extracted_docs() > 0:
```
Should this also check for ExtractedChemical records that don't already have related Ingredient records? I think the form is created and presented when the clean composition data has already been uploaded

Reply:
The ticket doesn't discuss this condition--I interpreted it to mean that the user could always upload a new .csv and overwrite the existing ingredient record